File: ELEC881_Copilot/src/Function Name Masked/Java script (JS)/q33-arithmetic-slices-ii-subsequence.js
var ab = function(nums) {
    let count = 0;
    let dp = new Array(nums.length).fill(0).map(() => new Map());
    for (let i = 0; i < nums.length; i++) {
        for (let j = 0; j < i; j++) {
            let diff = nums[i] - nums[j];
            let sum = dp[j].get(diff) || 0;
            count += sum;
            dp[i].set(diff, (dp[i].get(diff) || 0) + sum + 1);
        }
    }
    return count;
}

File: ELEC881_Copilot/src/Function Name Masked/Python/q33-arithmetic-slices-ii-subsequence.py
class ab(object):
    def abc(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        from collections import defaultdict
        if not nums:
            return 0
        dp = [defaultdict(int) for _ in range(len(nums))]
        res = 0
        for i in range(len(nums)):
            for j in range(i):
                diff = nums[i] - nums[j]
                dp[i][diff] += 1
                if diff in dp[j]:
                    dp[i][diff] += dp[j][diff]
                    res += dp[j][diff]
        return res